book/build-service.md

```diff
@@ -0,0 +1,160 @@
+
+`when` element could contain:
+- language
+  - Name of language from devfile .metadata.language, multiple languages can be defined: `java,nodejs,node`
+- componentName
+  - Name of Component CR
+- projectType
+  - projectType from devfile .metadata.projectType, can contain multiple values separated by a comma.
+- dockerfile
+  - if the Docker file is defined in generated devfile. Boolean value.
+- annotations
+  - match by Component CR annotations, expects map.
+- labels
+  - match by Component CR labels, expects map.
+
+There are a few parameters that are set in PipelineRun created by the Build Service:
+- git-url - taken from Component CR's spec.source.gitSource.url
+- revision - taken from Component CR's spec.source.gitSource.revision when it's defined, otherwise it's using the default branch of the git repository
+- output-image - generated by the Build Service the image repository is taken from Component CR's.
+  - for Default mode, the tag is `initial-build-$TIMESTAMP-$RANDOM_NUMBER"`
+  - for Custom mode, the tag is `{{ revision }}` which is evaluated by Pipelines As Code to commit SHA.
+- docker-file - path to Dockerfile, only used when dockerfile is detected in Component's devfile
+- path-context - context path for docker build, only used when dockerfile is detected in Component's devfile
+- skip-checks - set to 'true' when Component contains annotation `skip-initial-checks: true`
+
+### PaC PipelineRun Pruner Controller
+
+In Default mode the PipelineRuns are connected to Component CR using OwnerRef, which handles the deletion of the PipelineRun CR when Component CR is deleted.
+
+In Custom mode the PipelineRuns creation is handled by Pipelines as Code and there is no cleanup of the PipelineRuns when the Repository CR which manages the git repository is deleted.
+
+The purpose of the PaC PipelineRun Pruner Controller is to remove the PipelineRun CRs created for Component CR which is being deleted.
+
+### Git Tekton Resources Renovater
+
+In Custom mode the PipelineRun definition is created in the user's git repository under `.tekton` folder, this file is generated by Component Build Controller during Component CR creation. Git Tekton Resources Renovater is taking care of updating content of `.tekton` folder during the lifetime of the Component CR.
+
+Git Tekton Resources Renovater triggers the creation of Job CRs when BuildPipelineSelector `build-pipeline-selector` in `build-service` namespace is updated.
+
+Workflow:
+1. Get repositories installed for the GitHub Application matching to Components CRs and get tokens to access them.
```

Comment:
One more question .. Does the `renovate` GH App needs to be installed in the user's github repository in order to use this cleanup feature?

Reply:
No we are using global application which is used by Build-service and pipelines-as-code.

Reply:
Okay got it